src/coreclr/pal/src/thread/context.cpp

```diff
@@ -1858,6 +1858,17 @@ DBG_FlushInstructionCache(
         __builtin___clear_cache((char *)begin, (char *)endOrNextPageBegin);
         begin = endOrNextPageBegin;
     }
+#elif defined(HOST_RISCV64)
+    // __clear_cache() expanded from __builtin___clear_cache() is not implemented
+    // on Linux/RISC-V64, at least in GLIBC 2.36, and we have to make syscall directly.
+    //
+    // TODO-RISCV64: use __builtin___clear_cache() in future GLIBC versions
+
+#ifndef __NR_riscv_flush_icache
+    #define __NR_riscv_flush_icache 259
+#endif
```

Comment:
Think it is not needed.

Reply:
Similar logic used for other sys calls as well:
https://github.com/dotnet/runtime/blob/d638e1e012f2e15feb98eb1099fa33cec00e1768/src/coreclr/pal/src/thread/process.cpp#L67C1-L78C8